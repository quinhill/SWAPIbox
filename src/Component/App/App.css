.App {
  text-align: center;
  background-image: url('../../assets/death_star.jpeg');
  background-repeat: no-repeat;
  background-color: #000000;
  height: 100%;
  width: 100%;
}

