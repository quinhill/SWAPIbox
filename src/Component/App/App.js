import React, { Component } from 'react';
import './App.css';
import Header from '../Header/Header';
import ButtonContainer from '../ButtonContainer/ButtonContainer';
import CardContainer from '../CardContainer/CardContainer';
import {
  fetchCrawl, 
  fetchData
} from '../../helper/apiCaller';

class App extends Component {
  constructor() {
    super()

    this.state = {
      crawlData: {},
      data: []
    }
  }

  async componentDidMount() {
    const crawlData = await fetchCrawl()
    this.setState({crawlData})
  }

  getData = async (button) => {
    const data = await fetchData(button)
    this.setState({data})
  }

  render() {
    return (
      <div className="App">
        <Header />
        <ButtonContainer 
          getData={this.getData}
        />
        <CardContainer
          crawlData={this.state.crawlData}
          peopleData={this.state.peopleData}
          planetData={this.state.planetData}
          vehicleData={this.state.vehicleData}
        />
      </div>
    );
  }
}

export default App;
