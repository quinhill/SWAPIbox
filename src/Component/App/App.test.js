import React from 'react';
import App from './App';
import { shallow } from 'enzyme';

describe('App', () => {

  describe('getData', () => {

    let wrapper;
    let mockFetchData;
    let mockFetchCrawl;
    let mockDataArray;

     

    beforeEach(() => {
      mockFetchData = jest.fn()
      mockFetchCrawl = jest.fn()
      mockDataArray = [{}, {}]

      window.fetch = jest.fn().mockImplementation(() => 
                      Promise.resolve({json: () => 
                      Promise.resolve(mockDataArray)
                    }))

      wrapper = shallow(<App 
                          fetchData={mockFetchData}
                          fetchCrawl={mockFetchCrawl}
                          />)
    })

    it('should match snapshot', () => {
      expect(wrapper).toMatchSnapshot()
    })
  })
})
