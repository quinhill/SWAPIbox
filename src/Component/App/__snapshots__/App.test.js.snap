// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App getData should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    fetchCrawl={[Function]}
    fetchData={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Header />,
        <ButtonContainer
          checkState={[Function]}
/>,
        <CardContainer
          crawlData={Object {}}
          people={Array []}
          planets={Array []}
          type=""
          vehicles={Array []}
/>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "checkState": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "crawlData": Object {},
          "people": Array [],
          "planets": Array [],
          "type": "",
          "vehicles": Array [],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Header />,
          <ButtonContainer
            checkState={[Function]}
/>,
          <CardContainer
            crawlData={Object {}}
            people={Array []}
            planets={Array []}
            type=""
            vehicles={Array []}
/>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "checkState": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "crawlData": Object {},
            "people": Array [],
            "planets": Array [],
            "type": "",
            "vehicles": Array [],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
